DataCollector/CommandDataCollector.php

```diff
@@ -29,11 +30,21 @@ public function collect(Request $request, Response $response, ?Throwable $except
         $this->data = [
             'num_commands' => count($this->commandLogger),
             'commands' => array_map(
-                static function (Command $command) : string {
-                    return json_encode($command->getCommand());
+                static function (Command $command) : array {
```

Comment:
While writing a comment to suggest tests, I realised that the current test calls `collect`, but does not perform any assertion. May I ask you to add the assertions for the `getTime` method you added here? Feel free to ignore the command lists, but if we could have some test for the time logic, that would be great. Please let me know if this works for you.

Reply:
I tried to add tests for `CommandDataCollector` but `Command`, `CommandStartedEvent` or `CommandSucceededEvent` are final and have private properties. So, I don't know how to make tests work since we cannot instantiate or mock them.

Reply:
Maybe we can consider to use dg/bypass-finals. WDYT ?

Reply:
@alcaeus I added https://github.com/dg/bypass-finals to allow APM mocks.Is it OKay for you ?

Reply:
Sorry for the long delay. I've taken a look, and I don't think it's necessary to mock this. We don't need to check for the exact time, but inserting a document should get some commands logged so we can assert that the time is greater than 0 for all commands. Please let me know if you can make the changes or if you want me to take this over. Thanks for your work so far, this looks good apart from the testing!

Reply:
End of my holidays, I have updated tests to remove mocks !